datafusion/sqllogictest/test_files/sort_merge_join.slt

```diff
@@ -100,13 +100,14 @@ Alice 100 Alice 2
 Alice 50 Alice 1
 Alice 50 Alice 2
 
+# Uncomment when filtered RIGHT moved
```

Comment:
does this mean that if we merge this PR the answers are incorrect (aka that we will be introducing a regression for some period of time?)

Reply:
SMJ is experimental is disabled by default. Even now depending on data distribution every variant of filtered SMJ join is not stable and gives incorrect result, despite the fact that  simple tests passed, that's the reason the fuzz tests are commented out for filtered variants.

I understand the concern, but moving all variants to new approach will make the PR unmanageable to review. I'll try to move other variants as fast as possible, moreover the most popular variants(Inner, Left, LeftSemi) are moved in this PR

Reply:
The issues https://github.com/apache/datafusion/issues/11555
https://github.com/apache/datafusion/issues/12359 focus on SMJ correctness issues. Fuzz tests flaky because of SMJ has a problem do answering your question this PR is to make SMJ to work correctly